### PR TITLE
Add Terraform Support For Cloud SQL Preferred Zone Selection Feature

### PR DESCRIPTION
**Pull request by Khaledmohamedrefaat**
Add Terraform Support For Cloud SQL Preferred Zone Selection Feature

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [X] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
composer: added `database_config.zone` field in `google_composer_environment`
```

Fixes [hashicorp/terraform-provider-google/issues/16344](https://github.com/hashicorp/terraform-provider-google/issues/16344)

### CODE REVIEW

**Comment by PeWu** [comment]
LGTM

File: mmv1/third_party/terraform/services/composer/resource_composer_environment_test.go.erb
Comment (trodge) [inline mmv1/third_party/terraform/services/composer/resource_composer_environment_test.go.erb:1]: Please change [L463](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/services/composer/resource_composer_environment_test.go.erb#L463) to give the network a different name and avoid a 409 when this test runs in parallel with other tests.

**Review by trodge — CHANGES_REQUESTED** [review]
The error we're seeing in our tests is this:

```
Error: Error creating Network: googleapi: Error 409: The resource 'projects/ci-test-project-188019/global/networks/tf-test-composer-net-8380661580462607644' already exists, alreadyExists
```

I think this is caused by naming collision between different composer tests being run in parallel.

File: mmv1/third_party/terraform/services/composer/resource_composer_environment_test.go.erb
Comment (Khaledmohamedrefaat) [reply mmv1/third_party/terraform/services/composer/resource_composer_environment_test.go.erb:1]: The test should have a random of INT_MAX and the probability of this colliding is really low. Also, this PR doesn't change anything in the test.

I've re run the tests and it's passing.

Comment (trodge) [reply mmv1/third_party/terraform/services/composer/resource_composer_environment_test.go.erb:1]: I believe it collides with high probability if the tests are run simultaneously. I've opened an [issue](https://github.com/hashicorp/terraform-provider-google/issues/16380) to track a fix for this, but I think we can proceed with merging this PR.

**Review by trodge — APPROVED** [review]
LGTM, opened https://github.com/hashicorp/terraform-provider-google/issues/16380 for the test hiccup.

File: mmv1/third_party/terraform/go.mod.erb [hunk 1]
```diff
@@ -3,95 +3,101 @@ module github.com/hashicorp/terraform-provider-google
 go 1.19
 
 require (
-	cloud.google.com/go/bigtable v1.19.0
-	github.com/GoogleCloudPlatform/declarative-resource-client-library v1.52.0
+	cloud.google.com/go/bigtable v1.20.0
+	github.com/GoogleCloudPlatform/declarative-resource-client-library v1.54.0
```
Comment (c2thorn) [inline mmv1/third_party/terraform/go.mod.erb:7]: Was this PR intended to upgrade every available dependency?

This upgrades the DCL library and seems completely unrelated to DCL changes. Were the test failures from https://github.com/GoogleCloudPlatform/magic-modules/pull/9341#issuecomment-1780885768 addressed? 

I can only imagine potential side effects from upgrading all of these imports at once. @trodge

Comment (trodge) [reply mmv1/third_party/terraform/go.mod.erb:7]: I have reverted this PR.

Please open a new one that updates only the necessary dependencies.

**Pull request merged**